lib/clients/ClientsManager.js

```diff
@@ -0,0 +1,286 @@
+const CredentialsManager = require('../credentials/CredentialsManager');
+const BackbeatMetadataProxy = require('../BackbeatMetadataProxy');
+const { createBackbeatClient, createS3Client } = require('./utils');
+const { authTypeAssumeRole } = require('../constants');
+const { http: HttpAgent, https: HttpsAgent } = require('httpagent');
+
+const DELETE_INACTIVE_CREDENTIALS_INTERVAL = 1000 * 60 * 30; // 30m
+const MAX_INACTIVE_DURATION = 1000 * 60 * 60 * 2; // 2hr
```

Comment:
If these are role sessions, they will likely expire before this timeout? Shall we link each set of creds with their expiration date (updated each time we refresh them)?